src/storage/ExtensionBasedMapper.ts

```diff
@@ -0,0 +1,129 @@
+   *
+   * @returns Absolute path of the file.
+   */
+  public mapUrlToFilePath(identifier: ResourceIdentifier, id = ''): string {
+    return this.getAbsolutePath(this.getRelativePath(identifier), id);
+  }
+
+  /**
+   * Strips the rootFilepath path from the filepath and adds the baseRequestURI in front of it.
+   * @param path - The file path.
+   *
+   * @throws {@Link Error}
+   * If the file path does not match the rootFilepath path of the store.
+   *
+   * @returns Url of the file.
+   */
+  public mapFilePathToUrl(path: string): string {
+    if (!path.startsWith(this.rootFilepath)) {
+      throw new Error(`File ${path} is not part of the file storage at ${this.rootFilepath}.`);
+    }
+    return this.baseRequestURI + path.slice(this.rootFilepath.length);
+  }
+
+  /**
+   * Get the content type from a file path, using its extension.
+   * @param path - The file path.
+   *
+   * @returns Content type of the file.
+   */
+  public getContentTypeFromExtension(path: string): string {
+    const extension = /\.([^./]+)$/u.exec(path);
+    return (extension && this.types[extension[1].toLowerCase()]) || APPLICATION_OCTET_STREAM;
+  }
+
+  /**
+   * Get the absolute file path based on the rootFilepath of the store.
+   * @param path - The relative file path.
+   * @param identifier - Optional identifier to add to the path.
+   *
+   * @returns Absolute path of the file.
+   */
+  public getAbsolutePath(path: string, identifier = ''): string {
+    return joinPath(this.rootFilepath, path, identifier);
+  }
+
+  /**
+   * Strips the baseRequestURI from the identifier and checks if the stripped base URI matches the store's one.
+   * @param identifier - Incoming identifier.
+   *
+   * @throws {@link NotFoundHttpError}
+   * If the identifier does not match the baseRequestURI path of the store.
+   *
+   * @returns A string representing the relative path.
+   */
+  public getRelativePath(identifier: ResourceIdentifier): string {
+    if (!identifier.path.startsWith(this.baseRequestURI)) {
+      throw new NotFoundHttpError();
+    }
+    return identifier.path.slice(this.baseRequestURI.length);
+  }
+
+  /**
+   * Splits the identifier into the parent directory and its own file name/slug.
+   * If the identifier specifies a directory, slug will be undefined.
+   * @param identifier - Incoming identifier.
+   *
+   * @throws {@link ConflictHttpError}
+   * If the root identifier is passed.
+   *
+   * @returns A PathAbstraction object containing path and (optional) slug fields.
+   */
+  public extractSlug(identifier: ResourceIdentifier): PathAbstraction {
+    const [ , path, slug ] = /^(.*\/)([^/]+\/?)?$/u.exec(this.getRelativePath(identifier)) ?? [];
+    if ((typeof path !== 'string' || normalizePath(path) === '/') && typeof slug !== 'string') {
+      throw new ConflictHttpError('Container with that identifier already exists (root).');
+    }
+    return { path, slug };
```

Comment:
We don't normalize on return, is that expected?

Reply:
That's how it was in the file store, but there don't seem to be any problems when normalizing it, so let's just return the normalized path?

Reply:
Let's normalize indeed. Both path and slug, lilkely.